src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -52,35 +53,45 @@ trait Validation {
     override def apply(t: T): Result = if (!b(t)) Success else validator(t)
   }
 
-  implicit def every[T](implicit validator: Validator[T]): Validator[Iterable[T]] = {
-    new Validator[Iterable[T]] {
-      override def apply(seq: Iterable[T]): Result = {
-
-        val violations: Set[Violation] = seq match {
-          case m: Map[_, _] =>
-            m.map(item => (item, validator(item))).collect {
-              case ((k, v), f: Failure) => GroupViolation(k, "not valid", Some(s"($k)"), f.violations)
-            }(collection.breakOut)
-          case _ =>
-            seq.map(item => (item, validator(item))).zipWithIndex.collect {
-              case ((item, f: Failure), pos: Int) => GroupViolation(item, "not valid", Some(s"($pos)"), f.violations)
-            }(collection.breakOut)
+  implicit def everyKeyValue[T](implicit validator: Validator[T]): Validator[Iterable[(String, T)]] = {
+    def mapViolation(violation: Violation, desc: Description): Violation = {
+      violation match {
+        case RuleViolation(value, constraint, path) =>
+          RuleViolation(value, constraint, desc +: path)
+        case GroupViolation(value, constraint, children, path) =>
+          GroupViolation(value, constraint, children, desc +: path)
+      }
+    }
+    new Validator[Iterable[(String, T)]] {
```

Comment:
I thought wix accord has this already, no?

Reply:
It has the "each" operator, but it is insufficient for many cases.

It is fine for his case:

```
    // before
    portMappings is every(portMappingNetworkNameValidator(networks.flatMap(_.name)))
    // after
    portMappings.each is portMappingNetworkNameValidator(networks.flatMap(_.name))
```

It does not seem to work well for these cases:

```
      isTrue("Volume names must be unique") { (vols: Seq[AppVolume]) =>
        val names: Seq[String] = vols.collect{ case v: AppExternalVolume => v.external.name }.flatten
        names.distinct.size == names.size
      } and every(validVolume(container, enabledFeatures, secrets))

      update.uris is optional(every(api.v2.Validation.uriIsValid) and isTrue(
        "may not be set in conjunction with fetch") { (uris: Seq[String]) =>
          !(uris.nonEmpty && update.fetch.fold(false)(_.nonEmpty))
        })

```